src/main/resources/explanations/challenge41_reason.adoc

```diff
@@ -0,0 +1,9 @@
+*Why pre-hashing passwords is not a good idea?*
+
+Though pre-hashing passwords is an easy way to upgrade legacy hashes, it becomes more easier to crack if both the legacy and the newer hashes are available.
+
+In this case, we were able to reduce bcrypt hashes to insecure md5 hashes and then cracking it.
+
+The safest way to avoid this is to reset the passwords of all users and hashing the new passwords with the newer algorithm. But, this method is not user-friendly.
```

Comment:
```suggestion
The safest way to avoid this is to reset the passwords of all users and hash the new passwords with the newer algorithm. But, this method is not user-friendly.
```

Reply:
Can we maybe explain what password shucking means as well :) and relate to https://www.scottbrady91.com/authentication/beware-of-password-shucking :) ?